src/controllers/dynakube/dynakube_controller.go

```diff
@@ -174,183 +170,172 @@ func (controller *DynakubeController) reconcileIstio(dynakube *dynatracev1beta1.
 	return updated
 }
 
-func (controller *DynakubeController) reconcileDynaKube(ctx context.Context, dkState *status.DynakubeState, dkMapper *mapper.DynakubeMapper) {
-	dtcReconciler := DynatraceClientReconciler{
-		Client:              controller.client,
-		DynatraceClientFunc: controller.dtcBuildFunc,
-	}
-	dtc, upd, err := dtcReconciler.Reconcile(ctx, dkState.Instance)
-
-	dkState.Update(upd, "token conditions updated")
-	if dkState.Error(err) {
-		log.Error(err, "failed to check tokens")
-		return
-	}
+func (controller *DynakubeController) reconcileDynaKube(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) error {
 
-	dkState.ValidTokens = true
-	if !dtcReconciler.ValidTokens {
-		dkState.ValidTokens = false
-		dkState.Update(true, "tokens not valid")
-		return
+	dtcReconciler := NewDynatraceClientReconciler(controller.client, controller.dtcBuildFunc)
+	dtc, err := dtcReconciler.Reconcile(ctx, dynakube)
+	if err != nil {
+		log.Info("failed to create dynatrace client")
```

Comment:
do we have any guidelines when to wrap with `errors.WithMessage` and when to just `log.Info` and return the error? maybe I'm missing something, but it feels a bit arbitrary 😄

Reply:
we return the error, and it gets logged by the manager